src/DessserQCheck.ml

```diff
@@ -211,7 +211,9 @@ let rec shrink_value_type =
       let sv = shrink_maybe_nullable v in
       (sk >|= vt_of_mn) <+> (sv >|= vt_of_mn)
         <+>
-      (pair sk sv >|= fun (k, v) -> T.map k v)
+      (sk >|= fun k -> T.map k v)
+        <+>
+      (sv >|= fun v -> T.map k v)
```

Comment:
I think `Iter.pair` is better than only shrinking one of the pair element. What's your rationale ?

Reply:
My understanding is that first shrink only on the first then on the second is better as we try more cases.

For example if `(a, b)` is a bad value maybe `(shrink a, shrink b)` is not. But `(shrink a, b)` can be a bad value. So we try `(shrink shrink a, b)` and `(shrink a, shrink b)`. We test more cases but its slower.

Though just the other solution with Iter should be enough as if `(shrink a, shrink b)` is good we will return `(a, b)` which is worse than `(shrink a, b)` but could be enough.

Reply:
I cannot make sense of what you are saying.
Just use Iter.shrink.